### ISSUE TITLE
Controller fails to delete unregistered runner resources

### ISSUE THREAD

**Issue by boris-stoyanov** [summerwind/actions-runner-controller#12]
Hello,

Thank you for the project I ran into some problems when I tried to use a custom image for the runner as my use case requires it. Here is what I added:
```apiVersion: actions.summerwind.dev/v1alpha1
kind: Runner
metadata:
  name: testrepoansible-runner
spec:
  image: custom_container
  repository: testrepoansible
  env: []
```
I did this with 2 different containers, one had a typo and failed during pulling. The other did not have a github actions runner installed at all. The idea was to check if someone makes a mistake and adds a resource that does not exist what happens.

Running `kubectl delete runner testrepoansible-runner` , resource deleted, but the runner was still there. I checked the code and correct me if I am wrong but if the runner does not unregister it cannot be deleted. Here is the error from the controller:
```2020-03-19T11:51:06.185Z	ERROR	controllers.Runner	Failed to unregister runner	{"runner": "default/example-github-actions-runner", "error": "json: cannot unmarshal object into Go value of type []controllers.GitHubRunner"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/summerwind/actions-runner-controller/controllers.(*RunnerReconciler).Reconcile
	/workspace/controllers/runner_controller.go:100
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:256
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/wait/wait.go:88
2020-03-19T11:51:06.185Z	ERROR	controller-runtime.controller	Reconciler error	{"controller": "runner", "request": "default/example-github-actions-runner", "error": "json: cannot unmarshal object into Go value of type []controllers.GitHubRunner"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:258
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:232
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.4.0/pkg/internal/controller/controller.go:211
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/wait/wait.go:88
```
To remove the resource I had to patch the runner:
`kubectl patch runner example-github-actions-runner -p '{"metadata":{"finalizers":[]}}' --type=merge`

**Comment by mumoshu** [comment]
@boris-stoyanov Hey! Thanks for reporting. I believe this is a duplicate of #11 which should be fixed via #7.

Would you mind giving v0.4.0 a try which will be released shortly?

**Comment by boris-stoyanov** [comment]
Other than the scary long CRD everything is working fine with the new release.
